share/sysbuild/cmake/modules/sysbuild_images.cmake

```
@@ -5,6 +5,26 @@
 # This module is responsible for including images into sysbuild and to call
 # pre and post hooks.
 
+# Internal function to update build info with list of images abnd write the file.
+# Main reason for using an internal function is to properly scope variable usage.
+# Function takes a list of images.
+function(sysbuild_info_image images)
+  set(info_image_list)
+  foreach(image ${images})
+    ExternalProject_Get_Property(${image} SOURCE_DIR)
+    get_target_property(type ${image} APP_TYPE)
+    if(type)
+      list(APPEND info_image_list MAP "name: ${image}, source-dir: ${SOURCE_DIR}, type: ${type}")
+    else()
+      list(APPEND info_image_list MAP "name: ${image}, source-dir: ${source_dir}")
+    endif()
+  endforeach()
+  build_info(images VALUE ${info_image_list})
```

I think this should be a LIST?
```suggestion
  build_info(images LIST ${info_image_list})
```
I have mixed feelings on this. It should have been caught by `yaml_set` as an invalid syntax, but on the other hand, if `VALUE MAP` works on actual values (ie. maps, not map lists) it may offer a way to set multiple fields in one go?
Not blocking but looking for feedback / other opinions :slightly_smiling_face:

---

@pillo79 thanks for noticing, but the `build_info()` itself only supports VALUE and PATH, where PATH simply indicates that the value(s) should be adjusted to abs path:
https://github.com/zephyrproject-rtos/zephyr/blob/0c368e85b117272df8397d8258da088ebea11f5c/cmake/modules/extensions.cmake#L3700-L3703

The reason for that, is that `build_info()` uses the build_info schema to determine if the actual tag should be passed to `yaml_set()` as a list or a value.
https://github.com/zephyrproject-rtos/zephyr/blob/0c368e85b117272df8397d8258da088ebea11f5c/cmake/modules/extensions.cmake#L3759-L3767
That way the caller of `build_info()` doesn't need to consider this detail :wink: 

But this is only something we can do in build info, because we know there is a schema associated with build info.
The generic yaml implementation cannot know such details, and therefore requires the caller to specify this.